core/arch/arm32/include/kernel/thread.h

```diff
@@ -47,12 +49,15 @@ struct thread_smc_args {
 	uint32_t a6;	/* Not used */
 	uint32_t a7;	/* Hypervisor Client ID */
 };
+#endif /*ARM32*/
```

Comment:
Can't we remove  endif and ifdef? Or do you want each structure in the #ifdef?  (same below)